File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.ArrayList;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> arrayList = new ArrayList<>();
        arrayList.add(arr[0]); // 첫 번째 요소 추가

        for (int i = 1; i < arr.length; i++) {
            if (arr[i] != arr[i - 1]) { // 현재 값과 이전 값이 다를 때만 추가
                arrayList.add(arr[i]);
            }
        }

        int[] answer = new int[arrayList.size()];
        for (int i = 0; i < arrayList.size(); i++) {
            answer[i] = arrayList.get(i);
        }

        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    
    # 첫 번째 요소를 무조건 추가
    answer.append(arr[0])
    
    # 두 번째 요소부터 마지막 요소까지 순회
    for i in range(1, len(arr)):
        # 이전 요소와 현재 요소가 다르면 추가
        if arr[i] != arr[i-1]:
            answer.append(arr[i])
    
    return answer
